### PR TITLE
Rearrange "Users" sidebar menu

### PR DESCRIPTION
**Pull request by esoergel**
## Product Description

All user pages were filed under "Application Users" or "Project Users", but that dichotomy doesn't hold up very well today (not to mention the terminology is unorthodox).  Mobile workers can also have roles, and web users will soon be able to have user data, and in the future likely groups too.  This puts in a single "Users" section with both combined.

It also moves "Edit User Data" out of the mobile workers page and onto the sidebar as a top level item.

#### Before:
image

#### After:
image

## Technical Summary

https://dimagi-dev.atlassian.net/browse/USH-953

Pretty straightforward change technically.  I'm more concerned with the user-facing aspects of the change.  The URL for the user fields page does change, which is unfortunate, but I think should be fine.

### Safety story

Technically very straightforward change.  Happy with local testing

### QA Plan

N/A

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/tabs/tabclasses.py [hunk 1]
```diff
@@ -1716,13 +1701,16 @@ def _get_locations_menu(self):
...
         items = []
 
-        mobile_users_menu = self._get_mobile_users_menu()
-        if mobile_users_menu:
-            items.append((_('Application Users'), mobile_users_menu))
-
-        project_users_menu = self._get_project_users_menu()
-        if project_users_menu:
-            items.append((_('Project Users'), project_users_menu))
+        users_menu = filter(None, [
```
Comment (MartinRiese) [inline corehq/tabs/tabclasses.py:1709]: I really like this. So much cleaner.

File: corehq/tabs/tabclasses.py [hunk 2]
```diff
@@ -1527,6 +1525,13 @@ def _get_commcare_username(request=None, couch_user=None,
                 'show_in_dropdown': True,
             }
 
+    def _user_fields(self):
```
Comment (MartinRiese) [inline corehq/tabs/tabclasses.py:1528]: Do we have a guide line to do this in python over the template?

**Review by MartinRiese — APPROVED** [review]

**Comment by MartinRiese** [comment]
Going by your three questions:
Is this still safe to do? yes
Does this have user-facing changes we aren't aware of? I don't see any change other than the obvious ones.
Is this a backwards incompatible change? can be reverted. No change to the underlying data. so Yes

File: corehq/tabs/tabclasses.py (hunk 2 above)
Comment (esoergel) [reply corehq/tabs/tabclasses.py:1528]: I don't follow - do you mean generating the navigation bars?

Comment (MartinRiese) [reply corehq/tabs/tabclasses.py:1528]: Isn't this replacing?
```
{% if can_add_extra_users and can_access_all_locations %}
  <a class="btn btn-default" href="{% url "user_fields_view" domain %}" id="btn-edit_user_fields">
    <i class="fa fa-edit"></i>
    {% trans "Edit User Fields" %}
    </a>
{% endif %}
```

Comment (esoergel) [reply corehq/tabs/tabclasses.py:1528]: Oh I see - yes essentially this page was accessed via a button inside the mobile workers page, but this PR moves it a level up, so it is now accessed via the sidebar.  Now it's on the same level as the mobile workers page.  The sidebar and the navbar are controlled via this `tabclasses` file that manages the main tree of pages on HQ.

**Pull request merged**